Store clients UUIDs and apply restrictions - fixes #60 fixes #61

azixus opened the pull request:
Store the client UUID on connection and store his answered questions.
fixes #60
fixes #61

azixus committed:
Store uuid and corresponding socket i debate

azixus committed:
Merge branch 'dev' into fb-uuidValidations

azixus committed:
Log device uuid

azixus committed:
Multiple answers check in answerQuestion/answerOpenQuestion & tests

azixus committed:
Fixed Debate.getNbUniqueClients & tests

BecaudA commented on src/debate/debate.js:
```diff
@@ -163,6 +185,18 @@ export class Debate {
             return;
         }
 
+        if (question.isOpenQuestion) {
+            logger.debug(`Question with id (${questionId}) is not an open question.`);
+            callback(false);
+            return;
+        }
```
I assume that `question.isOpenQuestion` is set to `true` if the current question is an oppened one. Then why is the logger debuging 'Question with id (${questionId}) is **not** an open question.' if `question.isOpenQuestion` equal `true` ? \
Am I missing something ?

azixus replied:
It was a copy paste mistake. I fixed it in the latest commit.

``logger.debug(`Question with id (${questionId}) is an open question and not a closed question.`);``

azixus committed:
Fix logging in answerQuestion

Fix logging when the question is an open question instead of a closed question in answerQuestion.
